modules/core/src/main/scala/fs2/kafka/internal/syntax.scala

```diff
@@ -194,22 +195,24 @@ private[kafka] object syntax {
     def void: KafkaFuture[Unit] =
       map(_ => ())
 
-    def cancelToken[F[_]](implicit F: Sync[F]): CancelToken[F] =
-      F.delay { future.cancel(true); () }
+    def cancelToken[F[_]](implicit F: Async[F]): F[Option[F[Unit]]] =
+      F.blocking { future.cancel(true); () }.start.as(None)
```

Comment:
This looks strange with always returning `None`.

Reply:
The inner value is an optional cancellation token. You can't cancel the cancellation of the future, so that's why it's None - I think this is correct?

Reply:
Maybe the `as(None)` should move to the call site though

Reply:
Actually - we can return the cancel token for the cancelation fiber. updating.